src/systems/menu_defn.cpp

```diff
@@ -1289,22 +1329,123 @@ static Widget s_enabled_physics_widgets[] = {
             },
     },
     {
-        .type = WidgetType::Checkbox,
-        .checkbox =
+        .type = WidgetType::FloatEdit,
+        .float_edit =
             {
-                .label = "Moon Gravity",
-                .pref = pref::BoolPref::Moon,
+                .label = "Weight",
+                .pref = pref::U8Pref::Weight,
+                .precision = 1000,
+                .min = 0,
+                .max = 200,
+                .floor = 900,
+                .decimals = 3,
             },
     },
 };
 
+static const char* PHYSICS_PRESETS[] = {"Default",          "Light Ball",  "No Friction",
+                                        "Heavy Ball",       "Bouncy Ball", "Sticky Ball",
+                                        "Jump-Mod Physics", "Custom"};
+
+static Widget s_lightball[] = {{.type = WidgetType::Text, .text = {"  Weight: 1.00 -> 0.95"}}};
+static Widget s_nofriction[] = {{.type = WidgetType::Text, .text = {"  Friction: 0.010 -> 0.000"}}};
+static Widget s_heavyball[] = {{.type = WidgetType::Text, .text = {"  Weight: 1.00 -> 1.05"}}};
+static Widget s_bouncyball[] = {
+    {.type = WidgetType::Text, .text = {"  Restitution: 0.50 -> 1.20"}}};
+static Widget s_stickyball[] = {
+    {.type = WidgetType::Text, .text = {"  Restitution: 0.50 -> 0.01"}}};
+static Widget s_jump_physics[] = {
+    {.type = WidgetType::Text, .text = {"  Friction: 0.010 -> 0.015"}},
+    {.type = WidgetType::Text, .text = {"  Restitution: 0.50 -> 0.25"}},
+};
```

Comment:
Where these constants are defined, can you add a reminder comment to update this menu text? I totally get that it would be a PITA to read the constants in the menu programmatically (also it _is_ possible if you wanted...).

Reply:
You almost had me on board but I think I'd need to make another widget, I'll just add the comment for now

Reply:
You could add a `.text_func` function pointer field that's called if it's not `nullptr`. It could have type `void (*text_func)(char *buf)`, taking a buffer you `sprintf` into. No need to another widget type/flag/enum.

Reply:
Oh I had forgotten about the text functions! I'll just leave it how it is though because dealing with all those buffers does seem like a pain.